src/window.cpp

```diff
@@ -3121,21 +3121,20 @@ void Window::InvalidateData(int data, bool gui_scope)
 	this->SetDirty();
 	if (!gui_scope) {
 		/* Schedule GUI-scope invalidation for next redraw. */
-		this->scheduled_invalidation_data.push_back(data);
+		this->scheduled_invalidation_data = data;
```

Comment:
Given your explanation, might adding an `assert` here to ensure `scheduled_invalidation_data` is either equal to `data` or not set? Just to avoid future-us assuming it is possible, and breaking weird shit :D